commit cd7cdc550c2a00027d6ec1810d064ad636250b1f
Author: Peter Geoghegan <pg@bowt.ie>
Date:   Mon Apr 17 09:58:18 2023 -0700

    Fix incorrect comment about nbtree WAL record.
    
    The nbtree VACUUM WAL record stores its page offset number payload in
    blk 0 (just like the closely related nbtree DELETE WAL record).  Commit
    ebd551f5 fixed a similar issue with the DELETE WAL record, but missed
    this one.

diff --git a/src/include/access/nbtxlog.h b/src/include/access/nbtxlog.h
index a0310c96fb1..7dd9fd02065 100644
--- a/src/include/access/nbtxlog.h
+++ b/src/include/access/nbtxlog.h
@@ -199,11 +199,11 @@ typedef struct xl_btree_reuse_page
  * tuples on a leaf page.  The former variant is used by VACUUM, while the
  * latter variant is used by the ad-hoc deletions that sometimes take place
  * when btinsert() is called.
  *
  * The records are very similar.  The only difference is that xl_btree_delete
- * has a snapshotConflictHorizon field to generate recovery conflicts.
+ * have snapshotConflictHorizon/isCatalogRel fields for recovery conflicts.
  * (VACUUM operations can just rely on earlier conflicts generated during
  * pruning of the table whose TIDs the to-be-deleted index tuples point to.
  * There are also small differences between each REDO routine that we don't go
  * into here.)
  *
@@ -223,13 +223,17 @@ typedef struct xl_btree_reuse_page
 typedef struct xl_btree_vacuum
 {
 	uint16		ndeleted;
 	uint16		nupdated;
 
-	/* DELETED TARGET OFFSET NUMBERS FOLLOW */
-	/* UPDATED TARGET OFFSET NUMBERS FOLLOW */
-	/* UPDATED TUPLES METADATA (xl_btree_update) ARRAY FOLLOWS */
+	/*----
+	 * In payload of blk 0 :
+	 * - DELETED TARGET OFFSET NUMBERS
+	 * - UPDATED TARGET OFFSET NUMBERS
+	 * - UPDATED TUPLES METADATA (xl_btree_update) ITEMS
+	 *----
+	 */
 } xl_btree_vacuum;
 
 #define SizeOfBtreeVacuum	(offsetof(xl_btree_vacuum, nupdated) + sizeof(uint16))
 
 typedef struct xl_btree_delete
@@ -242,11 +246,11 @@ typedef struct xl_btree_delete
 
 	/*----
 	 * In payload of blk 0 :
 	 * - DELETED TARGET OFFSET NUMBERS
 	 * - UPDATED TARGET OFFSET NUMBERS
-	 * - UPDATED TUPLES METADATA (xl_btree_update) ARRAY
+	 * - UPDATED TUPLES METADATA (xl_btree_update) ITEMS
 	 *----
 	 */
 } xl_btree_delete;
 
 #define SizeOfBtreeDelete	(offsetof(xl_btree_delete, isCatalogRel) + sizeof(bool))
